statistics/histogram.go

```diff
@@ -969,14 +969,31 @@ func (hg *Histogram) OutOfRangeRowCount(sctx sessionctx.Context, lDatum, rDatum
 	}
 	rowCount = totalPercent * hg.NotNullCount()
 
+	// Upper bound logic
+
 	// Use the modifyCount as the upper bound. Note that modifyCount contains insert, delete and update. So this is
 	// a rather loose upper bound.
 	// There are some scenarios where we need to handle out-of-range estimation after both insert and delete happen.
 	// But we don't know how many increases are in the modifyCount. So we have to use this loose bound to ensure it
```

Comment:
it can be written.

```
optObjective := sctx.GetSessionVars().GetOptObjective()
if rowCount > float64(modifyCount) && optObjective != variable.OptObjectiveDetermined {
}
if optObjective ==. xxx {
}
```

Reply:
Updated.